File: src/main/java/net/minheur/potoflux/ui/dialogData/NewAccountData.java
package net.minheur.potoflux.ui.dialogData;

import net.minheur.potoflux.login.perms.Perms;

import java.util.List;

/**
 * New account data class.
 */
public class NewAccountData {
    /**
     * The email field.
     */
    public String email = "";
    /**
     * The first name field.
     */
    public String firstName = "";
    /**
     * The last name field.
     */
    public String lastName = "";
    /**
     * The password field.
     */
    public String password = "";
    /**
     * The rank field.
     */
    public Integer rank;
    /**
     * The perms field.
     */
    public List<Perms> perms;
}

File: src/main/kotlin/net/minheur/potoflux/ui/dialogData/NewAccountData.kt
package net.minheur.potoflux.ui.dialogData

import net.minheur.potoflux.login.perms.Perms

class NewAccountData {
    @JvmField
    var email: String = ""

    @JvmField
    var firstName: String = ""

    @JvmField
    var lastName: String = ""

    @JvmField
    var password: String = ""

    @JvmField
    var rank: Int? = null

    // Use MutableList so Java sees java.util.List<Perms> (not wildcard)
    @JvmField
    var perms: MutableList<Perms>? = null
}
